perception/autoware_map_based_prediction/include/map_based_prediction/map_based_prediction_node.hpp

```diff
@@ -291,6 +292,8 @@ class MapBasedPredictionNode : public rclcpp::Node
     const std::string & object_id);
   std::string tryMatchNewObjectToDisappeared(
     const std::string & object_id, std::unordered_map<std::string, TrackedObject> & current_users);
+  bool searchProperStartingRefPathIndex(
+    const TrackedObject & object, const PosePath & pose_path, size_t & index) const;
```

Comment:
can you change this function to return `optional<size_t>` as returned index value ?

Reply:
resolved https://github.com/autowarefoundation/autoware.universe/pull/8602/commits/f3bfec1d3ae3eb6bd09fc7a10b2d0176d3a4c420